pkg/utils/utils_test.go

```diff
@@ -0,0 +1,520 @@
+		t.Skipf("Directory symlinks not supported or permission denied: %v", err)
+	}
+
+	// Non-existent path
+	nonExistent := filepath.Join(tmpDir, "not_exists")
+
+	tests := []struct {
+		name             string
+		path             string
+		expectLink       bool
+		expectErr        bool
+		expectErrMessage string
+	}{
+		{
+			name:       "RegularFile",
+			path:       filePath,
+			expectLink: false,
+			expectErr:  false,
+		},
+		{
+			name:       "FileSymlink",
+			path:       symlinkPath,
+			expectLink: true,
+			expectErr:  false,
+		},
+		{
+			name:       "Directory",
+			path:       dirPath,
+			expectLink: false,
+			expectErr:  false,
+		},
+		{
+			name:       "DirectorySymlink",
+			path:       dirSymlinkPath,
+			expectLink: true,
+			expectErr:  false,
+		},
+		{
+			name:       "NonExistent",
+			path:       nonExistent,
+			expectLink: false,
+			expectErr:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			isLink, err := IsPathSymlink(tt.path)
+			if isLink != tt.expectLink {
+				t.Errorf("Expected isLink=%v, got %v", tt.expectLink, isLink)
+			}
+			if (err != nil) != tt.expectErr {
+				t.Errorf("Expected error=%v, got %v", tt.expectErr, err)
+			}
+		})
+	}
+}
+
+func TestCreateSymlink(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	// Create a target file
+	targetFile := filepath.Join(tmpDir, "target.txt")
+	if err := os.WriteFile(targetFile, []byte("hello"), 0644); err != nil {
+		t.Fatalf("Failed to create target file: %v", err)
+	}
+	fileLink := filepath.Join(tmpDir, "file_link.txt")
+
+	// Create a target directory
+	targetDir := filepath.Join(tmpDir, "target_dir")
+	if err := os.Mkdir(targetDir, 0755); err != nil {
+		t.Fatalf("Failed to create target directory: %v", err)
+	}
+	dirLink := filepath.Join(tmpDir, "dir_link")
+
+	tests := []struct {
+		name     string
+		link     string
+		target   string
+		isDir    bool
+		wantErr  bool
+		validate func(t *testing.T, linkPath string)
+	}{
+		{
+			name:    "FileSymlink",
+			link:    fileLink,
+			target:  targetFile,
+			isDir:   false,
+			wantErr: false,
+			validate: func(t *testing.T, linkPath string) {
+				fi, err := os.Lstat(linkPath)
+				if err != nil {
+					t.Fatalf("Symlink not created: %v", err)
+				}
+				if fi.Mode()&os.ModeSymlink == 0 {
+					t.Errorf("Expected symlink, got mode: %v", fi.Mode())
+				}
+			},
+		},
+		{
+			name:    "DirSymlink",
+			link:    dirLink,
+			target:  targetDir,
+			isDir:   true,
+			wantErr: false,
+			validate: func(t *testing.T, linkPath string) {
+				fi, err := os.Lstat(linkPath)
+				if err != nil {
+					t.Fatalf("Symlink not created: %v", err)
+				}
+				if fi.Mode()&os.ModeSymlink == 0 {
+					t.Errorf("Expected symlink, got mode: %v", fi.Mode())
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := CreateSymlink(tt.link, tt.target, tt.isDir)
+			if (err != nil) != tt.wantErr {
```

Comment:
What about having two ifs for this part, collapsing two checks into one makes it harder to read.